### PR TITLE
docs: update vault k8s to 0.3.0

### PR DESCRIPTION
**Pull request by jasonodonnell**
This updates the Vault K8s docs with the new features added in 0.3.0.  Additionally I broke out the annotations to a separate webpage as the list is growing and needed better navigation for users.

### CODE REVIEW

File: website/pages/docs/platform/k8s/injector/annotations.mdx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  by adding the value `injected` to the pod after a successful mutation.
+
+- `vault.hashicorp.com/agent-configmap` - name of the configuration map where Vault
+  Agent configuration file and templates can be found.
+
+- `vault.hashicorp.com/agent-image` - name of the Vault docker image to use. This
+  value overrides the default image configured in the controller and is usually
+  not needed. Defaults to `vault:1.3.2`.
+
+- `vault.hashicorp.com/agent-init-first` - configures the pod to run the Vault Agent
```
Comment (kalafut) [inline website/pages/docs/platform/k8s/injector/annotations.mdx]: I was slightly confused about his parameter in the k8s PR. You mentioned in the changes that it allows the init container to run first "or last". So if this is false does it always run last?

Comment (jasonodonnell) [reply website/pages/docs/platform/k8s/injector/annotations.mdx]: That's correct, I can clarify.

Comment (jasonodonnell) [reply website/pages/docs/platform/k8s/injector/annotations.mdx]: Done!

File: website/pages/docs/platform/k8s/injector/annotations.mdx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+- `vault.hashicorp.com/agent-inject-template` - configures the template Vault Agent
+  should use for rendering a secret. The name of the template is any
+  unique string after `vault.hashicorp.com/agent-inject-template-`, such as
+  `vault.hashicorp.com/agent-inject-template-foobar`. This should map to the same
+  unique value provided in `vault.hashicorp.com/agent-inject-secret-`. If not provided,
+  a default generic template is used.
+
+- `vault.hashicorp.com/agent-inject-token` - configures Vault Agent to share the Vault 
+  token with other containers in the mode.  This is helpful when other containers
```
Comment (kalafut) [inline website/pages/docs/platform/k8s/injector/annotations.mdx]:
```suggestion
  token with other containers in the pod.  This is helpful when other containers 
```

**Pull request merged**